Apps/Examples/Examples/All Examples/SkyLayerExample.swift

```diff
@@ -0,0 +1,124 @@
+        let azimuthalAngle: Double = 0
+
+        // Indicates the sun's position relative to the horizon. A value of `90` places the sun's center at the
+        // horizon line. Lower values place the sun below the horizon line, while higher values place the sun's
+        // center further above the horizon line.
+        let polarAngle: Double = 90
+        skyLayer.skyAtmosphereSun = .constant([azimuthalAngle, polarAngle])
+
+        // The intensity or brightness of the sun.
+        skyLayer.skyAtmosphereSunIntensity = .constant(10)
+
+        // Set the sky's color to light blue with a light pink halo effect.
+        skyLayer.skyAtmosphereColor = .constant(ColorRepresentable(color: .skyBlue))
+        skyLayer.skyAtmosphereHaloColor = .constant(ColorRepresentable(color: .lightPink))
+
+        try! mapView.mapboxMap.style.addLayer(skyLayer)
+    }
+
+    // Update the sky type when the `UISegmentedControl` value is changed.
+    @objc func updateSkyLayer() {
+        var skyType: Value<SkyType>
+        if segmentedControl.selectedSegmentIndex == 0 {
+            skyType = .constant(.gradient)
+        } else {
+            skyType = .constant(.atmosphere)
+        }
+
+        // Update the sky layer based on the updated segmented control value.
+        try! mapView.mapboxMap.style.updateLayer(withId: skyLayer.id) { (layer: inout SkyLayer) throws in
+            layer.skyType = skyType
+        }
+    }
+
+    func addTerrainLayer() {
+        // Add a RasterDEMSource. This will be used to create and add a terrain layer. r
+        var demSource = RasterDemSource()
+        demSource.url = "mapbox://mapbox.mapbox-terrain-dem-v1"
+        demSource.tileSize = 512
+        demSource.maxzoom = 14.0
+        try! mapView.mapboxMap.style.addSource(demSource, id: "mapbox-dem")
+
+        var terrain = Terrain(sourceId: "mapbox-dem")
+        terrain.exaggeration = .constant(1.5)
+
+        try! mapView.mapboxMap.style.setTerrain(terrain)
+    }
+
+    func addSegmentedControl() {
+        segmentedControl = UISegmentedControl(items: ["Gradient", "Atmosphere"])
+        segmentedControl.backgroundColor = .lightGray
+        segmentedControl.selectedSegmentIndex = 0
+        segmentedControl.addTarget(self, action: #selector(updateSkyLayer), for: .valueChanged)
+        view.insertSubview(segmentedControl, aboveSubview: mapView)
+
+        segmentedControl.translatesAutoresizingMaskIntoConstraints = false
+        let safeArea = view.safeAreaLayoutGuide
+        NSLayoutConstraint.activate([
+            segmentedControl.bottomAnchor.constraint(equalTo: safeArea.bottomAnchor, constant: -80),
+            segmentedControl.centerXAnchor.constraint(equalTo: safeArea.centerXAnchor)
+        ])
+    }
+
+    override public func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+         // The below line is used for internal testing purposes only.
+        finish()
```

Comment:
this `finish()` should probably move to the `onNext(.styleLoaded)` closure above.